code/modules/shuttle/shuttle.dm

```diff
@@ -280,24 +288,54 @@
 
 /// Called when a new shuttle arrives
 /obj/docking_port/stationary/proc/on_arrival(obj/docking_port/mobile/arriving_shuttle)
+	turn_off_landing_lights()
 	return
 
 /// Called when a new shuttle is about to arrive
 /obj/docking_port/stationary/proc/on_prearrival(obj/docking_port/mobile/arriving_shuttle)
+	log_debug("SHUTTLE DEBUG: [name] static prearrival called!")
+	turn_on_landing_lights()
 	return
 
 /// Called when the docked shuttle ignites
 /obj/docking_port/stationary/proc/on_dock_ignition(obj/docking_port/mobile/departing_shuttle)
+	turn_on_landing_lights()
 	return
 
 /// Called when a shuttle is about to complete undocking from this stationary dock (already physically gone)
 /obj/docking_port/stationary/proc/on_departure(obj/docking_port/mobile/departing_shuttle)
+	turn_off_landing_lights()
 	return
```

Comment:
You'll want to make sure everything calls parent here cause some don't

Reply:
I don't see anywhere not calling parent, or at least not anywhere that would affect the changes made

Reply:
Examples where implementations of these procs don't get called by parent:
`/obj/docking_port/stationary/emergency_response/external/on_arrival(obj/docking_port/mobile/arriving_shuttle)`
`/obj/docking_port/stationary/lifeboat_dock/on_dock_ignition(departing_shuttle)`
`/obj/docking_port/stationary/emergency_response/external/on_arrival(obj/docking_port/mobile/arriving_shuttle)`

Also if you want linters to enforce we always call parent you can add `SHOULD_CALL_PARENT(TRUE)` beneath any proc definition.